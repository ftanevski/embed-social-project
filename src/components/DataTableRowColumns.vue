<template>
	<div class="col-md-2">
		<p class="text-muted" data-testid="title-test">
			{{ titleToUpperCase }}
		</p>
        <div class="font-w600" data-testid="text-test">
            {{ text }}
        </div>
	</div>
</template>

<script>
export default {
	name: 'DataTableRowColumns',
	props: {
		title: {
            type: String,
			required: true,
			description: 'String that defines the key from the key: value properties of the submissions'
		},
		text: {
            type: String,
			required: true,
			description: 'String that defines the value of the key: value properties from the submissions'
		}
	},
	computed: {
		titleToUpperCase() {
            return this.title.charAt(0).toUpperCase() + this.title.slice(1, this.title.length)
		} 
	}
}
</script>