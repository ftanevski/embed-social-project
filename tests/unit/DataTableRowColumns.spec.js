import { mount } from '@vue/test-utils';
import DataTableRowColumns from '../../src/components/DataTableRowColumns';

describe('DataTableRowColumns', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = mount(DataTableRowColumns, {
			props: {
				title: 'name',
				text: 'Jonathan'
			}
		})
	})

	test('Snapshot matches', () => {
		expect(wrapper.element).toMatchSnapshot()
	})

	test('Computed property titleToUpperCase works with current prop value', () => {
		expect(wrapper.vm.titleToUpperCase).toBe('Name')
	})

	test('Computed property titleToUpperCase works with different prop value', async () => {
		await wrapper.setProps({ title: 'test' })
		expect(wrapper.vm.titleToUpperCase).toBe('Test')
	})
})