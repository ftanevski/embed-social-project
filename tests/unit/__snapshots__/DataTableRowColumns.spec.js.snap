// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableRowColumns Snapshot matches 1`] = `
<div
  class="col-md-2"
>
  <p
    class="text-muted"
  >
    Name
  </p>
  <div
    class="font-w600"
  >
    Jonathan
  </div>
</div>
`;
